src/host-config/proxy.ts

```diff
@@ -138,6 +184,32 @@ export class RedsocksConf {
 	}
 }
 
+function dnsToDnsu2t(
+	dns: unknown,
```

Comment:
should this be `boolean | string` instead?

Reply:
if we validate the input before passing it to the proxy, then this can just receive a `DnsConfig` as input

Reply:
Input is validated by making the dns input type more strict [here](https://github.com/balena-os/balena-supervisor/pull/2284/commits/39efabcccc28605b2ba3b538d48cf37b6cee0dcd#diff-4a1d4485db76d595a8f99491b2dbfbde15e43e6151f3bc3f2d16edcd29129ce7R4-R30) (from `string` to regex-bounded `AddressString`), which is used when calling `hostConfig.parse` [here](https://github.com/balena-os/balena-supervisor/pull/2284/commits/a774b95c97d29fe51fe1f4c6e7a5ee435936143e#diff-aaac9f5c9b6707e5fe6448bad4487ef43f9686b75b9728aebc50aab095e0c4f4R35-R63), throwing an error message which is tested [here](https://github.com/balena-os/balena-supervisor/pull/2284/commits/a774b95c97d29fe51fe1f4c6e7a5ee435936143e#diff-d3f5db585e205253dbabde1162d4c39c17be636d36c5a1354f7ae90583ff4cf3L839-R898).